### PR TITLE
Add support for the Allocs profiles produced by Julia's Allocs Profiler

### PR DESCRIPTION
**Pull request by NHDaly**
Add support for the Allocs profiles produced by Julia's Allocs Profiler

Support for visualizing the results from the allocations profiler in
draft PR:
JuliaLang/julia#42768.

This was basically copy/pasted from
https://github.com/vilterp/AllocProfileParser.jl.

Screen Shot 2022-01-03 at 9 50 41 PM
Screen Shot 2022-01-03 at 9 45 39 PM

### CODE REVIEW

File: src/allocs_profile.jl [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+using PProf.perftools.profiles: ValueType, Sample, Function, Location, Line, Label
+using PProf: _enter!
+const PProfile = PProf.perftools.profiles.Profile
+using Base.StackTraces: StackFrame
+
+using PProf.ProtoBuf
+using PProf.OrderedCollections
+
+# input: e.g. "maybe_handle_const_call! at ./compiler/ssair/inlining.jl:1243"
+function parse_location(loc_str::String)
```
Comment (vilterp) [inline src/allocs_profile.jl:23]: not used

**Review by vilterp — APPROVED** [review]
nice, thanks!

**Comment by vchuravy** [comment]
Now that this has landed in Base we should land this

**Comment by NHDaly** [comment]
Agreed. Just made some fixups to the API; this is ready to go now!

**Review by vilterp — APPROVED** [review]
re-approving 😛

File: test/Allocs.jl [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        # Profile compilation
+        @eval foo(x,y) = x * y + y / x
+        @eval foo(2, 3)
+    end
+
+    # Write the profile
+    outf = PProf.Allocs.pprof(out=out, web=false)
+
+    # Read the exported profile
+    prof = open(io->readproto(io, PProf.perftools.profiles.Profile()), outf, "r")
```
Comment (vilterp) [inline test/Allocs.jl:23]: 😮 what is this `->` syntax??

Comment (vchuravy) [reply test/Allocs.jl:23]: Lambda/Closure/Anonymous Function

Comment (vilterp) [reply test/Allocs.jl:23]: oh, of course 🤦‍♂️ just hadn't seen it without parens I guess

**Comment by NHDaly** [comment]
Cool! Thanks all - merging now

**Comment by NHDaly** [comment]
(leaving the branch up for anyone who had pointed their PProf at it locally)

**Pull request merged**